monkey/monkey_island/cc/ui/src/components/pages/RunMonkeyPage/RunManually/LocalManualRunOptions.js

```diff
@@ -28,29 +29,55 @@ const getContents = (props) => {
   const [osType, setOsType] = useState(OS_TYPES.WINDOWS_64);
   const [selectedIp, setSelectedIp] = useState(props.ips[0]);
   const [commands, setCommands] = useState(generateCommands());
+  const [customUsername, setCustomUsername] = useState('');
+  const customUsernameRef = useRef(null);
```

Comment:
Why do we need both of these? We use ref to persist username between os/ip changes? In that case why do we need a state for `customUsername`? As far as I can tell this value is only used to trigger a re-rendering, which makes the code confusing.

Reply:
You're right, we don't need ref